### ISSUE TITLE
bug: SQS queue name defaults to region in Resource Browser

### ISSUE THREAD

**Issue by iyoung-tixr** [localstack/localstack#9532]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If I create an SQS queue in the LocalStack Web Application, I can see it in the list.  However, the value in the Name column is the name of the region.  If I click on that, I get a "the specified queue does not exist for this wsdl version" error, understandably, because there is no queue by that name.  If I change the URL in the browser's address bar to reflect the actual queue name, it displays the queue's details without any error.

### Expected Behavior

The link in the list of queues should show the queue name, not the region name.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker-compose.yml

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    Open the LocalStack Web Application and go to Resource Browser -> SQS.  Create a queue and, when the new queue appears in the list, click on it.

### Environment

```markdown
- OS: macOS Ventura 13.5
- LocalStack: 2.3.2
- LocalStack Web Application: 0.3.1
```

### Anything else?

<img width="1308" alt="Screenshot 2023-11-01 at 19 20 53" src="https://github.com/localstack/localstack/assets/117106195/5ad51b5a-83bb-4c17-99aa-d0d6e165d94b">
<img width="1319" alt="Screenshot 2023-11-01 at 19 21 08" src="https://github.com/localstack/localstack/assets/117106195/025866a6-bf66-426c-8a2c-06a0912ce534">

Simplified docker-compose.yml for reference:
```yaml
version: '3.7'
services:
  localstack:
    container_name: localstack
    image: localstack/localstack
    dns:
      - 172.20.0.20
    networks:
      tixr:
        ipv4_address: 172.20.0.20
    ports:
      - 4566:4566
      - 4510-4559:4510-4559
      - 8000:4566
    environment:
      - DEBUG=${DEBUG-}
      - SQS_ENDPOINT_STRATEGY=path
      - MAIN_CONTAINER_NAME=localstack
      - AWS_ACCESS_KEY_ID=000000000000
      - AWS_SECRET_ACCESS_KEY=test
networks:
  tixr:
    name: tixr
    ipam:
      config:
        - subnet: 172.20.0.0/24
```

**Comment by jusmark123** [comment]
I am experiencing the same after pulling the latest docker image on both Mac and Linux. Also, getting curl 6 error with the new SQS queueUrls, I had to set SQS_ENDPOINT_STRATEGY to "path" to get it working, and now the queue name is the region in the SQS interface.

**Comment by mattviasat** [comment]
I am possibly seeing something similar. I am using the latest boto3 on the `latest` localstack ( also tried on 2.3) and no matter how I configure it to use `us-west-2` I get `us-east-1`.


from boto stream logging. Note what I request and what I get.
```
DEBUG: Sending http request: <AWSPreparedRequest stream_output=False, method=POST, url=http://sqs.us-west-2.localstack:4566/, headers={'X-Amz-Target': b'AmazonSQS.CreateQueue', 'Content-Type': b'application/x-amz-json-1.0', 'User-Agent': b'Boto3/1.29.2 md/Botocore#1.32.2 ua/2.0 os/linux#6.1.0-13-cloud-amd64 md/arch#x86_64 lang/python#3.8.10 md/pyimpl#CPython cfg/retry-mode#legacy Botocore/1.32.2', 'X-Amz-Date': b'20231117T160738Z', 'Authorization': b'AWS4-HMAC-SHA256 Credential=/20231117/us-west-2/sqs/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-target, Signature=...', 'amz-sdk-invocation-id': b'eed20afc-770a-4d5c-9bf3-ed058fca89c1', 'amz-sdk-request': b'attempt=1', 'Content-Length': '35'}>
2023-11-17 16:07:38,294 botocore.endpoint [DEBUG] Sending http request: <AWSPreparedRequest stream_output=False, method=POST, url=http://sqs.us-west-2.localstack:4566/, headers={'X-Amz-Target': b'AmazonSQS.CreateQueue', 'Content-Type': b'application/x-amz-json-1.0', 'User-Agent': b'Boto3/1.29.2 md/Botocore#1.32.2 ua/2.0 os/linux#6.1.0-13-cloud-amd64 md/arch#x86_64 lang/python#3.8.10 md/pyimpl#CPython cfg/retry-mode#legacy Botocore/1.32.2', 'X-Amz-Date': b'20231117T160738Z', 'Authorization': b'AWS4-HMAC-SHA256 Credential=/20231117/us-west-2/sqs/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-target, Signature=...', 'amz-sdk-invocation-id': b'eed20afc-770a-4d5c-9bf3-ed058fca89c1', 'amz-sdk-request': b'attempt=1', 'Content-Length': '35'}>
DEBUG: Starting new HTTP connection (1): sqs.us-west-2.localstack:4566
2023-11-17 16:07:38,295 urllib3.connectionpool [DEBUG] Starting new HTTP connection (1): sqs.us-west-2.localstack:4566
DEBUG: http://sqs.us-west-2.localstack:4566 "POST / HTTP/1.1" 200 84
2023-11-17 16:07:38,299 urllib3.connectionpool [DEBUG] http://sqs.us-west-2.localstack:4566 "POST / HTTP/1.1" 200 84
DEBUG: Response headers: {'Content-Type': 'application/x-amz-json-1.0', 'Content-Length': '84', 'x-amzn-requestid': '9b07674a-11b4-4f75-bdc1-e6ab3c318030', 'Connection': 'close', 'date': 'Fri, 17 Nov 2023 16:07:38 GMT', 'server': 'hypercorn-h11'}
2023-11-17 16:07:38,299 botocore.parsers [DEBUG] Response headers: {'Content-Type': 'application/x-amz-json-1.0', 'Content-Length': '84', 'x-amzn-requestid': '9b07674a-11b4-4f75-bdc1-e6ab3c318030', 'Connection': 'close', 'date': 'Fri, 17 Nov 2023 16:07:38 GMT', 'server': 'hypercorn-h11'}
DEBUG: Response body:
b'{"QueueUrl": "http://sqs.us-east-1.localstack:4566/000000000000/cat-queue-3216c986"}'
2023-11-17 16:07:38,299 botocore.parsers [DEBUG] Response body:
b'{"QueueUrl": "http://sqs.us-east-1.localstack:4566/000000000000/cat-queue-3216c986"}'
DEBUG: Event needs-retry.sqs.CreateQueue: calling handler <botocore.retryhandler.RetryHandler object at 0x7f3c4fdb0d90>
```


Note how `WS sns.CreateTopic => 200; 000000000000/us-east-1;` but the ``
```
 INFO --- [   asgi_gw_0] localstack.request.aws     : AWS sns.CreateTopic => 200; 000000000000/us-east-1; CreateTopicInput({'Name': 'cat-topic-d50ba680'}, headers={'Host': 'localstack:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'Boto3/1.29.2 md/Botocore#1.32.2 ua/2.0 os/linux#6.1.0-13-cloud-amd64 md/arch#x86_64 lang/python#3.8.10 md/pyimpl#CPython cfg/retry-mode#legacy Botocore/1.32.2', 'X-Amz-Date': '20231117T155411Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=/20231117/us-west-2/sns/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=14ad94f73634ae85b8fb679fdbec4d8fcc8737f177d2d7b188d45359e15cb591', 'amz-sdk-invocation-id': 'd172a3ba-c36a-4075-9d77-f19e2a2cde90', 'amz-sdk-request': 'attempt=1', 'Content-Length': '61', 'x-localstack-tgt-api': 'sns', 'x-moto-account-id': '000000000000'}); CreateTopicResponse({'TopicArn': 'arn:aws:sns:us-west-2:000000000000:cat-topic-d50ba680'}, headers={'Content-Type': 'text/xml', 'Content-Length': '340', 'Connection': 'close'})
```
```
DEBUG --- [   asgi_gw_1] l.aws.serving.wsgi         : POST sqs.us-west-2.localstack:4566/
DEBUG --- [   asgi_gw_1] l.services.sqs.provider    : creating queue key=cat-queue-d50ba680 attributes=None tags=None
DEBUG --- [   asgi_gw_1] l.aws.protocol.serializer  : No accept header given. Using request's Content-Type (application/x-amz-json-1.0) as preferred response Content-Type.
 INFO --- [   asgi_gw_1] localstack.request.aws     : AWS sqs.CreateQueue => 200; 000000000000/us-east-1; CreateQueueRequest({'QueueName': 'cat-queue-d50ba680'}, headers={'Host': 'sqs.us-west-2.localstack:4566', 'Accept-Encoding': 'identity', 'X-Amz-Target': 'AmazonSQS.CreateQueue', 'Content-Type': 'application/x-amz-json-1.0', 'User-Agent': 'Boto3/1.29.2 md/Botocore#1.32.2 ua/2.0 os/linux#6.1.0-13-cloud-amd64 md/arch#x86_64 lang/python#3.8.10 md/pyimpl#CPython cfg/retry-mode#legacy Botocore/1.32.2', 'X-Amz-Date': '20231117T155411Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=/20231117/us-west-2/sqs/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-target, Signature=ed321cbc0e0dbe79f6243d9d0a748c2824b50ebb851221b6a72580251d709f5c', 'amz-sdk-invocation-id': '3e441e36-dbda-4c48-a5c2-cd20a0c1ffe7', 'amz-sdk-request': 'attempt=1', 'Content-Length': '35', 'x-localstack-tgt-api': 'sqs', 'x-moto-account-id': '000000000000'}); CreateQueueResult({'QueueUrl': 'http://sqs.us-east-1.localstack:4566/000000000000/cat-queue-d50ba680'}, headers={'Content-Type': 'application/x-amz-json-1.0', 'Content-Length': '84', 'x-amzn-requestid': '538e558b-ea64-4b51-a5bb-746f2fe78b24', 'Connection': 'close'})
DEBUG --- [   asgi_gw_0] localstack.aws.accounts    : No Account ID in thread-local storage for thread 139840981038784
DEBUG --- [   asgi_gw_1] l.aws.serving.wsgi         : POST localstack:4566/
DEBUG --- [   asgi_gw_1] l.aws.protocol.serializer  : No accept header given. Using request's Content-Type (application/x-www-form-urlencoded; charset=utf-8) as preferred response Content-Type.
 INFO --- [   asgi_gw_1] localstack.request.aws     : AWS sns.Subscribe => 200; 000000000000/us-east-1; SubscribeInput({'TopicArn': 'arn:aws:sns:us-west-2:000000000000:cat-topic-d50ba680', 'Protocol': 'sqs', 'Endpoint': 'arn:aws:sqs:us-west-2:000000000000:cat-queue-d50ba680'}, headers={'Host': 'localstack:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'Boto3/1.29.2 md/Botocore#1.32.2 ua/2.0 os/linux#6.1.0-13-cloud-amd64 md/arch#x86_64 lang/python#3.8.10 md/pyimpl#CPython cfg/retry-mode#legacy Botocore/1.32.2', 'X-Amz-Date': '20231117T155413Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=/20231117/us-west-2/sns/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=4dfaf63d7c7260cb546dc24cf17036ab0c59a1104eea0e3171a148a555082942', 'amz-sdk-invocation-id': '34738ac2-fb37-4f73-acc1-f880525e6714', 'amz-sdk-request': 'attempt=1', 'Content-Length': '194', 'x-localstack-tgt-api': 'sns', 'x-moto-account-id': '000000000000'}); SubscribeResponse({'SubscriptionArn': 'arn:aws:sns:us-west-2:000000000000:cat-topic-d50ba680:5050e488-21d1-4205-8040-77ea4e200502'}, headers={'Content-Type': 'text/xml', 'Content-Length': '383', 'Connection': 'close'})
DEBUG --- [   asgi_gw_0] l.aws.serving.wsgi         : POST localstack:4566/
ERROR --- [   asgi_gw_0] l.aws.handlers.logging     : exception during call chain
Traceback (most recent call last):
  File "/opt/code/localstack/localstack/aws/chain.py", line 90, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack/aws/handlers/service.py", line 123, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack/aws/handlers/service.py", line 93, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 166, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/forwarder.py", line 135, in _call
    return handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 118, in __call__
    return self.fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/services/sns/provider.py", line 537, in publish
    publish_ctx.topic_attributes |= vars(topic_model)
                                    ^^^^^^^^^^^^^^^^^
TypeError: vars() argument must have __dict__ attribute
DEBUG --- [   asgi_gw_0] l.aws.protocol.serializer  : No accept header given. Using request's Content-Type (application/x-www-form-urlencoded; charset=utf-8) as preferred response Content-Type.
 INFO --- [   asgi_gw_0] localstack.request.aws     : AWS sns.Publish => 500 (InternalError); 000000000000/us-east-1; PublishInput({'TopicArn': 'arn:aws:sns:us-west-2:000000000000:cat-topic-d50ba680', 'Message': '1', 'MessageAttributes': {'string': {'DataType': 'String', 'StringValue': 'String'}}}, headers={'Host': 'localstack:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'Boto3/1.29.2 md/Botocore#1.32.2 ua/2.0 os/linux#6.1.0-13-cloud-amd64 md/arch#x86_64 lang/python#3.8.10 md/pyimpl#CPython cfg/retry-mode#legacy Botocore/1.32.2', 'X-Amz-Date': '20231117T155414Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=/20231117/us-west-2/sns/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=9ec848d8f692be981308bdd7b6ec22d2c25a8b6f9ed445dd2d27e8eea94acf5e', 'amz-sdk-invocation-id': '5742e78b-b266-4837-b94b-7ac71499c16d', 'amz-sdk-request': 'attempt=1', 'Content-Length': '253', 'x-localstack-tgt-api': 'sns', 'x-moto-account-id': '000000000000'}); InternalError(exception while calling sns.Publish: Traceback (most recent call last):
  File "/opt/code/localstack/localstack/aws/chain.py", line 90, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack/aws/handlers/service.py", line 123, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack/aws/handlers/service.py", line 93, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 166, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/forwarder.py", line 135, in _call
    return handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 118, in __call__
    return self.fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack/services/sns/provider.py", line 537, in publish
    publish_ctx.topic_attributes |= vars(topic_model)
                                    ^^^^^^^^^^^^^^^^^
TypeError: vars() argument must have __dict__ attribute
, headers={'Content-Type': 'text/xml', 'Content-Length': '1651', 'Connection': 'close'})
LocalStack supervisor: Shutting down localstack (PID 15)...

```

**Comment by dogabudak** [comment]
I think we are seeing the same issue, AWS_REGION is not passed accurately.

**Comment by bentsku** [comment]
@mattviasat I think the issue is a bit different, I can help you through it.

First, the SNS issue should now be resolved with #9679.

For you region issue, this comes from the following:
`'Authorization': b'AWS4-HMAC-SHA256 Credential=/20231117/us-west-2/sqs/aws4_request`
Your authorization header is malformed, it does not have an access key id, so LocalStack fails to parse it, and will default back to the default account id and the default region `us-east-1`. 

You would need to configure your boto client this way:
```python
    config = Config(
        region_name="eu-west-2"
    )
    sqs_client = boto3.client(
        service_name="sqs",
        endpoint_url="<your-endpoint-here>",
        config=config,
        aws_access_key_id="test",
        aws_secret_access_key="test",
    )
```

This is all explained in the following issue: #9701, in this comment: https://github.com/localstack/localstack/issues/9701#issuecomment-1821583480

This is a different issue than from the original issue, which has to do with the parsing of the SQS queue in our Resource Browser.

**Comment by thrau** [comment]
thanks for reporting Ian @iyoung-tixr. We confirmed the issue and have a fix in the pipeline. Should be released in a couple of days.
